cpp/src/arrow/compute/kernels/scalar_if_else_test.cc

```diff
@@ -316,5 +313,98 @@ TEST_F(TestIfElseKernel, IfElseDispatchBest) {
   CheckDispatchBest(name, {null(), uint8(), int8()}, {boolean(), int16(), int16()});
 }
 
+template <typename Type>
+class TestIfElseBaseBinary : public ::testing::Test {};
+
+TYPED_TEST_SUITE(TestIfElseBaseBinary, BinaryTypes);
+
+TYPED_TEST(TestIfElseBaseBinary, IfElseBaseBinary) {
```

Comment:
IMO these cases could be consolidated into one or two checks since there are only 12 cases here (or 5 if you want to prune: null cond, true cond with null/non-null left array, false cond with null/non-null right array).

Reply:
well, there are 64 cases TBH. Each param can be null/ non-null and scalar/ array (4 variations). Since we have 3 params we have 4*4*4 = 64 permutations. :slightly_smiling_face:

Reply:
`CheckWithDifferentShapes` actually checks for all those cases + with offsets. But I agree, in `CheckWithDifferentShapes` it could check similar cases redundantly in several iterations. But we decided to leave it as it is.